pkg/providers/vsphere/vmprovider_vm_test.go

```diff
@@ -332,10 +332,12 @@ func vmTests() {
 
 			Context("GetVirtualMachineProperties", func() {
 				const (
-					propName        = "config.name"
-					propPowerState  = "runtime.powerState"
-					propExtraConfig = "config.extraConfig"
-					propPathName    = "config.files.vmPathName"
+					propName              = "config.name"
+					propPowerState        = "runtime.powerState"
+					propExtraConfig       = "config.extraConfig"
+					propPathName          = "config.files.vmPathName"
+					propExtraConfigKeyKey = "vmservice.example"
+					propExtraConfigKey    = `config.extraConfig["` + propExtraConfigKeyKey + `"]`
```

Comment:
Note that govmomi has `mo.Field` type now, which is mainly used on vcsim side, but can simplify client-side too:
```go
field := mo.Field{Path: propExtraConfig, Key: propExtraConfigKeyKey}
prop := field.String() // `config.extraConfig["vmservice.example"]`
```